pennylane/_device.py

```diff
@@ -656,10 +656,12 @@ def default_expand_fn(self, circuit, max_expansion=10):
         """
         # pylint: disable=protected-access
 
-        comp_basis_sampled_multi_measure = (
-            len(circuit.measurements) > 1 and circuit.samples_computational_basis
+        multi_measure_comp_basis_without_obs = len(circuit.measurements) > 1 and any(
+            m.measures_computational_basis and m.obs is None for m in circuit.measurements
```

Comment:
We need to add this `any` statement to keep the same functionality as before (expand the tape only when `obs` is `None`).